akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/persistence/EnhancedPersistentActor.scala

```diff
@@ -37,24 +37,73 @@ abstract class EnhancedPersistentActor[E:ClassTag, Ex <: Exception : ClassTag]
 
   implicit val ec = context.dispatcher
 
-  private   var isProcessingEvent = false
-  private   var pendingDurableMessage:Option[DurableMessage] = None
-  private   var timeoutTimer:Option[Cancellable] = None
+  case class InternalState(
+                            // Used when processing events live - not recovering
+                            var eventLogLevelInfo: Boolean,
+
+                            // Used when recovering events
+                            var recoveringEventLogLevelInfo: Boolean,
+
+                            // Used when processing commands
+                            var cmdLogLevelInfo: Boolean,
+
+                            // Will be set using the correct logLevel when starting to do something
+                            var currentLogLevelInfo: Boolean,
+                            var prevLogLevelTryCommand: Boolean,
+
+                            var persistAndApplyEventHasBeenCalled: Boolean,
+                            var dmGeneratingVersionFixedDeliveryIds: Set[Long],
 
-  // Used when processing events live - not recovering
-  private   var eventLogLevelInfo = true
+                            var currentDmGeneratingVersion: Int,
+                            var addDmGeneratingVersionIfSavingEvents: Boolean,

```

Comment:
also resolved runtime

Reply:
tips... when moving props out again of InternalState, It's a good idea to "revert" the code so that you get as little diff as possible.... :)

Reply:
For sure, thanks for the help.
What about the three I left out, can some of them go in?

Reply:
I think it was correct to leave them out